presto-kafka/src/main/java/com/facebook/presto/kafka/KafkaConsumerManager.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.kafka;
+
+import com.google.common.cache.CacheBuilder;
+import com.google.common.cache.CacheLoader;
+import com.google.common.cache.LoadingCache;
+import io.airlift.log.Logger;
+import org.apache.kafka.clients.consumer.ConsumerConfig;
+import org.apache.kafka.clients.consumer.KafkaConsumer;
+import org.apache.kafka.common.serialization.ByteBufferDeserializer;
+
+import javax.annotation.PreDestroy;
+import javax.inject.Inject;
+
+import java.util.Map;
+import java.util.Properties;
+
+import static java.util.Objects.requireNonNull;
+
+/**
+ * Manages connections to the Kafka nodes. A worker may connect to multiple Kafka nodes depending on partitions
+ * it needs to process.
+ */
+public class KafkaConsumerManager
+{
+    private static final Logger log = Logger.get(KafkaConsumerManager.class);
+    public final LoadingCache<KafkaThreadIdentifier, KafkaConsumer> consumerCache;
+    private final int maxPartitionFetchBytes;
+    private final int maxPollRecords;
+
+    @Inject
+    public KafkaConsumerManager(KafkaConnectorConfig kafkaConnectorConfig)
+    {
+        requireNonNull(kafkaConnectorConfig, "kafkaConfig is null");
+        this.maxPartitionFetchBytes = kafkaConnectorConfig.getMaxPartitionFetchBytes();
+
+        this.consumerCache = CacheBuilder.newBuilder().build(CacheLoader.from(this::createConsumer));
+        this.maxPollRecords = kafkaConnectorConfig.getMaxPollRecords();
+    }
+
+    @PreDestroy
+    public void tearDown()
+    {
+        for (Map.Entry<KafkaThreadIdentifier, KafkaConsumer> entry : consumerCache.asMap().entrySet()) {
+            try {
+                entry.getValue().close();
+                consumerCache.invalidate(entry.getKey());
+            }
+            catch (Exception e) {
+                log.warn(e, "While closing consumer %s:", entry.getKey());
+            }
+        }
+    }
+
+    public KafkaConsumer getConsumer(KafkaThreadIdentifier consumerId)
```

Comment:
We are not really using this method, looks like we create a new consumer every time in each thread.  If so, I think we don't need the Loading Cache.

Reply:
Good point. Removed the unused cache.